app/src/main/java/com/example/mobiledger/data/repository/TransactionRepository.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.mobiledger.data.repository
+
+import com.example.mobiledger.data.sources.api.model.UserApi
+import com.example.mobiledger.domain.AppResult
+import com.example.mobiledger.domain.entities.MonthlyTransactionSummaryEntity
+import com.example.mobiledger.domain.entities.TransactionEntity
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+interface TransactionRepository {
+    suspend fun getMonthlyTransactionFromFirebaseDb(uid: String, monthYear: String): AppResult<MonthlyTransactionSummaryEntity?>
+    suspend fun addUserTransactionToFirebase(
+        uid: String,
+        monthYear: String,
+        transactionId: String,
+        transaction: TransactionEntity,
+        monthlyTransactionSummaryEntity: MonthlyTransactionSummaryEntity?
+    ): Boolean
+}
+
```

Comment:
Update Transaction id to time in millis from counters

Reply:
Done